client/src/Stores/MicrosoftLearnFilter.store.ts

```diff
@@ -0,0 +1,99 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import _ from "lodash";
+import { action, observable } from "mobx";
+import { getfiltersToDisplay, loadFiltersFromQueryParams, loadExpandedProductsFromQueryParams, getUpdatedURIfromSelectedFilters } from "../Features/MicrosoftLearn/MicrosoftLearnFilterCore";
+import { Catalog, LearnContent, Product } from "../Models/Learn";
+import { FilterType } from "../Models/Learn/FilterType.model";
+import { ChildStore } from "./Core";
+import { Filter } from "../Models/Learn/Filter.model";
+import * as H from 'history';
+
+export class MicrosoftLearnFilterStore extends ChildStore{
+  @observable displayFilter: Filter = new Filter();
+  @observable selectedFilter: Filter = new Filter();
+  @observable learnFilterUriParam: string = '';
+  @observable expandedProducts: string[]=[];
+
+  productMap: Map<string, Product>  = new Map<string, Product>(); 
+
+  @action
+  initializeFilters(catalog: Catalog, filterParams: URLSearchParams | undefined): void{
+    this.productMap = catalog.products;
+    //this.getProductHierarchicalMap(catalog);
+    
+    if(filterParams){
+      this.selectedFilter = loadFiltersFromQueryParams(filterParams, this.productMap);
+      this.expandedProducts = loadExpandedProductsFromQueryParams(filterParams);
+      this.learnFilterUriParam = getUpdatedURIfromSelectedFilters(this.selectedFilter, this.expandedProducts, this.productMap);        
+    }
+  }
+
+  @action
+  updateSearchTerm(newTerm: string, history: H.History): void{
+    this.selectedFilter.terms = newTerm.split(' ')
+    this.clone();
+    this.updateHistory(history);
+  }
+
+  @action
+  updateExpandedProducts(action: boolean, id: string, history: H.History): void{
+    action? this.expandedProducts.push(id) :
+      this.expandedProducts = this.expandedProducts.filter(pId => pId!==id);                
+    this.updateHistory(history);
+  }
+  
+  @action
+  addFilter(type: FilterType, filters: string[], history: H.History): void{
+    this.selectedFilter.set(type, [...this.selectedFilter.get(type), ...filters]);
+    this.clone();
+    this.updateHistory(history);
+  }
+
+  @action
+  removeFilter(type: FilterType, filters: string[], history: H.History): void{
+    this.selectedFilter.set(type, this.selectedFilter.get(type).filter(item => !filters.includes(item)));
+    this.clone();
+    this.updateHistory(history);
+  }
+
+  @action
+  resetFilter(history: H.History): void{
+    [FilterType.products, FilterType.roles, FilterType.types, FilterType.levels, FilterType.terms]
+      .forEach(type => this.selectedFilter.set(type, []));
+    this.clone();
+    this.updateHistory(history);
+  }
+
+  public updateFiltersToDisplay(catalog: Catalog | null, filteredCatalogContent: LearnContent[] | null): Filter{
```

Comment:
I'm a bit confused about this wrapper? Is not possible to change the call-site to never pass-in `filteredCatalogContent = null` (then we can get rid of this redirection) ?